src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/services/DraftAdjudicationService.kt

```diff
@@ -49,42 +49,42 @@ class DraftAdjudicationService(
   }
 
   @Transactional
-  fun addIncidentStatement(id: Long, statement: String): DraftAdjudicationDto {
-    val draftAdjudication =
-      draftAdjudicationRepository.findById(id)
-        .orElseThrow { throwEntityNotFoundException(id) }
+  fun addIncidentStatement(id: Long, statement: String?, completed: Boolean?): DraftAdjudicationDto {
+    throwIfStatementAndCompletedIsNull(statement, completed)
+
+    val draftAdjudication = draftAdjudicationRepository.findById(id).orElseThrow { throwEntityNotFoundException(id) }
 
     if (draftAdjudication.incidentStatement != null)
       throw IllegalStateException("DraftAdjudication already includes the incident statement")
 
-    draftAdjudication.incidentStatement = IncidentStatement(statement = statement)
+    draftAdjudication.incidentStatement = IncidentStatement(statement = statement, completed = completed)
 
     return draftAdjudicationRepository.save(draftAdjudication).toDto()
   }
 
-  fun throwEntityNotFoundException(id: Long): Nothing =
-    throw EntityNotFoundException("DraftAdjudication not found for $id")
-
   @Transactional
   fun editIncidentDetails(id: Long, locationId: Long?, dateTimeOfIncident: LocalDateTime?): DraftAdjudicationDto {
     val draftAdjudication = draftAdjudicationRepository.findById(id).orElseThrow { throwEntityNotFoundException(id) }
 
-    locationId?.let { draftAdjudication.incidentDetails?.locationId = it }
-    dateTimeOfIncident?.let { draftAdjudication.incidentDetails?.dateTimeOfIncident = it }
+    locationId?.let { draftAdjudication.incidentDetails.locationId = it }
+    dateTimeOfIncident?.let { draftAdjudication.incidentDetails.dateTimeOfIncident = it }
 
     return draftAdjudicationRepository
       .save(draftAdjudication)
       .toDto()
   }
 
   @Transactional
-  fun editIncidentStatement(id: Long, statement: String): DraftAdjudicationDto {
+  fun editIncidentStatement(id: Long, statement: String?, completed: Boolean?): DraftAdjudicationDto {
+    throwIfStatementAndCompletedIsNull(statement, completed)
```

Comment:
I take it that this can never be null? We don't allow them to write the statement and hit save without having selected the completed / not completed radio.

Reply:
Actually, you're right it's never null. I was thinking of the scenario where you just want to mark the statement complete. But now I think about it the front end will always post the incident statement.